src/pages/workspaces/workspace/tools/WorkflowView.js

```
@@ -88,50 +103,63 @@ class WorkflowView extends Component {
       _.keys(await workspace.entities())
     )
 
-    const { invalidInputs, invalidOutputs, methodConfiguration: config } =
-      await workspace.methodConfig(workflowNamespace, workflowName).validate()
+    const validationResponse = await workspace.methodConfig(workflowNamespace, workflowName).validate()
+    const { methodConfiguration: config } = validationResponse
+    const ioDefinitions = await Rawls.methodConfigInputsOutputs(config)
 
-    const processIO = _.flow(
-      _.update('inputs', preprocessIOList),
-      _.update('outputs', preprocessIOList)
-    )
-
-    const processedIO = processIO(await Rawls.methodConfigInputsOutputs(config))
+    const inputsOutputs = this.createIOLists(validationResponse, ioDefinitions)
 
     this.setState({
-      inputsOutputs: processedIO,
-      invalid: this.createInvalidIOMap(invalidInputs, invalidOutputs, config, processedIO),
-      config, entityTypes
+      config, entityTypes, inputsOutputs, ioDefinitions,
+      anyInvalidInput: _.some('error', inputsOutputs.inputs),
+      anyInvalidOutput: _.some('error', inputsOutputs.outputs)
     })
   }
 
+  createIOLists(validationResponse, ioDefinitions = this.state.ioDefinitions) {
+    const { invalidInputs, invalidOutputs, methodConfiguration: config } = validationResponse
+
+    const invalid = {
+      inputs: invalidInputs,
+      outputs: invalidOutputs
+    }
+
+    const process = ioKey => _.map(({ name, inputType, outputType, optional }) => {
+      const value = config[ioKey][name]
+      const [task, variable] = _.takeRight(2, _.split('.', name))
+      const type = (inputType || outputType).match(/(.*?)\??$/)[1] // unify, and strip off trailing '?'
+      const error = Utils.cond(
+        [optional, () => undefined],
```

Not 100% sure, but this looks like it might end up squashing server-reported validation errors on optional fields. Can you double check this?

---

It definitely is. The validation is returning errors for empty optional values, which is wrong, so FC was simply ignoring all errors on optional fields. However, that seems wrong so maybe we can do something better here.